package/kedro_viz/api/graphql.py

```diff
@@ -99,6 +93,92 @@ def get_all_runs() -> List[Run]:
     return runs
 
 
+def format_run_tracking_data(tracking_data: Dict) -> JSONObject:
+    """Convert tracking data in the front-end format.
+    [{
+        datasetName: 'Data Analysis',
+        datasetType: "kedro.extras.datasets.tracking.metrics_dataset.MetricsDataSet"
+        data: {
+            bootstrap: [
+            { runId: 'My Favorite Sprint', value: 0.8 },
+            { runId: 'Another favorite sprint', value: 0.5 },
+            { runId: 'Slick test this one', value: 1 },
+            ],
+            classWeight: [
+            { runId: 'My Favorite Sprint', value: 23 },
+            { runId: 'Another favorite sprint', value: 21 },
+            { runId: 'Slick test this one', value: 21 },
+            ]
+    }]
+
+    Args:
+        tracking_data: JSON blob of tracking data for selected runs
+    Returns:
+        Dictionary with formatted tracking data for selected runs
+    """
+    tracking_keys = set()
+    for key in tracking_data.keys():
+        for nested_keys in tracking_data[key].keys():
+            tracking_keys.add(nested_keys)
+    runs_tracking_data = {
+        key: [
+            {"runId": run_id, "value": tracking_data[run_id][key]}
+            for run_id in tracking_data
+            if key in tracking_data[run_id]
+        ]
+        for key in sorted(tracking_keys)
+    }
+    return json.loads(json.dumps(runs_tracking_data))
+
+
+def get_run_tracking_data(run_ids: List[ID]) -> List[TrackingDataSet]:
+    # pylint: disable=protected-access,import-outside-toplevel
+    """Get all details for a specific run. Run details contains the data from the
+    tracking MetricsDataSet and JSONDataSet instances that have been logged
+    during that specific `kedro run`.
+    Args:
+        run_ids:  List of IDs of runs to fetch the details for.
+
+    Returns:
+        List of TrackingDataSets
+
+    """
+    from kedro.extras.datasets.tracking import JSONDataSet, MetricsDataSet  # noqa: F811
+
+    all_datasets = []
+    catalog = data_access_manager.catalog.get_catalog()
+    experiment_datasets = [
+        (ds_name, ds_value)
+        for ds_name, ds_value in catalog._data_sets.items()
+        if (isinstance(ds_value, (MetricsDataSet, JSONDataSet)))
+    ]
+
+    if not experiment_datasets:
+        logger.warning("No tracking datasets found in catalog")
+
+    for name, dataset in experiment_datasets:
+        all_runs = {}
+        for run_id in run_ids:
+            runid = ID(run_id)
+            file_path = dataset._get_versioned_path(str(runid))
+            if Path(file_path).is_file():
+                with dataset._fs.open(
+                    file_path, **dataset._fs_open_args_load
+                ) as fs_file:
+                    json_data = json.load(fs_file)
+                    all_runs[runid] = json_data
```

Comment:
Could you add a comment here explaining why we need to rely on low-level I/O methods and not being able to utilise `catalog.load`?

Also I thought we agree to abstract this bit out so we could re-use it?